x/concentrated-liquidity/incentives_test.go

```diff
@@ -2987,3 +2970,175 @@ func (s *KeeperTestSuite) TestCreateIncentive() {
+			// we recalibrate the position accum value after claiming).
+			positionAccumDelta := positionPost.InitAccumValue.Sub(positionPre.InitAccumValue)
+			s.Require().Equal(tc.growthInside, positionAccumDelta)
+		})
+	}
+}
+
+// Note that the non-forfeit cases are thoroughly tested in `TestCollectIncentives`
+func (s *KeeperTestSuite) TestClaimAllIncentives() {
+	uptimeHelper := getExpectedUptimes()
+	defaultSender := s.TestAccs[0]
+	tests := map[string]struct {
+		name              string
+		poolId            uint64
+		growthInside      []sdk.DecCoins
+		growthOutside     []sdk.DecCoins
+		forfeitIncentives bool
+		expectError       error
+	}{
+		"happy path: claim rewards without forfeiting": {
+			growthInside:  uptimeHelper.hundredTokensMultiDenom,
+			growthOutside: uptimeHelper.twoHundredTokensMultiDenom,
+		},
+		"claim and forfeit rewards": {
+			growthInside:      uptimeHelper.hundredTokensMultiDenom,
+			growthOutside:     uptimeHelper.twoHundredTokensMultiDenom,
+			forfeitIncentives: true,
+		},
+		"claim and forfeit rewards when no rewards have accrued": {
+			forfeitIncentives: true,
+		},
+		"claim and forfeit rewards with varying amounts and different denoms": {
+			growthInside:      uptimeHelper.varyingTokensMultiDenom,
+			growthOutside:     uptimeHelper.varyingTokensSingleDenom,
+			forfeitIncentives: true,
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		s.Run(tc.name, func() {
+			// Setup test env.
+			s.SetupTest()
+			clPool := s.PrepareConcentratedPool()
+			clKeeper := s.App.ConcentratedLiquidityKeeper
+
+			// Initialize position.
+			err := clKeeper.InitOrUpdatePosition(s.Ctx, validPoolId, defaultSender, DefaultLowerTick, DefaultUpperTick, sdk.OneDec(), s.Ctx.BlockTime(), time.Hour*24*14)
+			s.Require().NoError(err)
+
+			clPool.SetCurrentTick(DefaultCurrTick)
+			if tc.growthOutside != nil {
+				s.addUptimeGrowthOutsideRange(s.Ctx, validPoolId, defaultSender, DefaultCurrTick.Int64(), DefaultLowerTick, DefaultUpperTick, tc.growthOutside)
+			}
+
+			if tc.growthInside != nil {
+				s.addUptimeGrowthInsideRange(s.Ctx, validPoolId, defaultSender, DefaultCurrTick.Int64(), DefaultLowerTick, DefaultUpperTick, tc.growthInside)
+			}
+
+			err = clKeeper.SetPool(s.Ctx, clPool)
+			s.Require().NoError(err)
+
+			// Store initial accum values for comparison later
+			initUptimeAccumValues, err := clKeeper.GetUptimeAccumulatorValues(s.Ctx, validPoolId)
+			s.Require().NoError(err)
+
+			// Get newly created position to pass into sut function
+			position, err := clKeeper.GetPosition(s.Ctx, validPoolId, defaultSender, DefaultLowerTick, DefaultUpperTick, s.Ctx.BlockTime(), time.Hour*24*14)
+
+			// System under test.
+			amountClaimed, err := clKeeper.ClaimAllIncentivesForPosition(s.Ctx, validPoolId, defaultSender, position, DefaultLowerTick, DefaultUpperTick, tc.forfeitIncentives)
+
+			if tc.expectError != nil {
+				s.Require().Error(err)
+				s.Require().ErrorIs(err, tc.expectError)
```

Comment:
I don't believe we have any error cases we are testing here, can we:

1. Verify that there aren't any error cases we need to check here and
2. If we add error cases, change this to ErrorContains. Otherwise, remove.

I believe there are likely error cases we want to test on ClaimAllIncentivesForPosition though

Reply:
Since this function was extracted from `CollectIncentives`, there are (thorough) tests there for all the relevant cases on `ClaimAllIncentivesForPosition` (see comment above tests), but I agree that at least having sanity error checks is good even if duplicates.

Adding rn

Reply:
fd86d6a598ae3654fe93e91a2ea331a673237980